compiler/rustc_macros/src/hash_stable.rs

```diff
@@ -64,6 +64,50 @@ pub(crate) fn hash_stable_generic_derive(
     )
 }
 
+pub(crate) fn hash_stable_no_context_derive(
```

Comment:
Can we have an inner function with a bool to indicate whether to add the where clause or not, then just delegate these two derives to that?

Reply:
Sure

Reply:
Cool, r=me with that change.